### PR TITLE
Add Python 3.10 to CI tests

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

Add Python 3.10 to our test suite.

## Motivation and Context

Good to keep testing against the latest Python version.

## How Has This Been Tested?

Letting CI run against it.

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by RishiDiwanTT** [comment]
@tdilauro Please do not merge into `main` yet as the webpub library points to a repo branch at the moment

**Comment by tdilauro** [comment]
> @tdilauro Please do not merge into `main` yet as the webpub library points to a repo branch at the moment

Okay. And you have it in `draft` status, which is a good way to ensure that it is NOT merged. 👍🏽

File: pyproject.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ multipledispatch = "0.6.0"
 nameparser = "1.1.0"  # nameparser is for author name manipulations
 nltk = "3.6.7"  # nltk is a textblob dependency.
 oauth2client = "4.1.3"  # Deprecated and should be replaced.
-palace-webpub-manifest-parser = "~2.0.3"
+palace-webpub-manifest-parser = { git = "https://github.com/ThePalaceProject/webpub-manifest-parser.git", branch = "feature/py310" }
```
Comment (tdilauro) [inline pyproject.toml:45]: Was this used for testing? Is there a py 3.10-compatible version of `palace-webpub-manifest-parser` up on PyPI that we can use here, instead?

**Review by tdilauro — COMMENTED** [review]
Mostly looks good, but we're currently requiring a feature branch, rather than a version, for the web pub manifest parser.

**Comment by jonathangreen** [comment]
@RishiDiwanTT new version of webpub-manifest parser is release on PyPi now, so you should be able to pull that in here.
https://pypi.org/project/palace-webpub-manifest-parser/

File: pyproject.toml (hunk 1 above)
Comment (RishiDiwanTT) [reply pyproject.toml:45]: Yup the webpub PR was still in progress yesterday. It is updated now in PyPi Jonathan

**Review by RishiDiwanTT — APPROVED** [review]

**Pull request merged**